src/parachain/escrow.ts

```diff
@@ -146,29 +146,72 @@ export class DefaultEscrowAPI extends DefaultTransactionAPI implements EscrowAPI
     async totalVotingSupply(
         blockNumber?: number
     ): Promise<MonetaryAmount<Currency<VoteUnit>, VoteUnit>> {
-        const [currentBlockNumber, epoch, span, rawSlopeChanges] = await Promise.all([
-            this.systemAPI.getCurrentBlockNumber(),
-            this.api.query.escrow.epoch(),
+        let block;
+        let epoch;
+
+        if (blockNumber) {
+            block = new BN(blockNumber);
+            const maxEpoch = await this.api.query.escrow.epoch();
+            epoch = await this.findBlockEpoch(block, maxEpoch.toBn());
```

Comment:
The `maxEpoch` query could be moved inside `findBlockEpoch` since that function is already async. I don't think the caller should be aware of how to fetch the maxEpoch